qiskit_experiments/library/characterization/resonator_spectroscopy.py

```diff
@@ -175,26 +175,38 @@ def __init__(
 
         Raises:
             QiskitError: If no frequencies are given and absolute frequencies are desired and
-                no backend is given.
+                no backend is given or the backend does not have default measurement frequencies.
         """
         analysis = ResonatorSpectroscopyAnalysis()
 
         if frequencies is None:
             frequencies = np.linspace(-20.0e6, 20.0e6, 51)
 
             if absolute:
-                if backend is None:
-                    raise QiskitError(
-                        "Cannot automatically compute absolute frequencies without a backend."
-                    )
-
-                center_freq = BackendData(backend).meas_freqs[physical_qubits[0]]
-                frequencies += center_freq
+                frequencies += self._get_backend_meas_freq(
+                    BackendData(backend) if backend is not None else None,
+                    physical_qubits[0],
+                )
 
         super().__init__(
             physical_qubits, frequencies, backend, absolute, analysis, **experiment_options
         )
 
+    @staticmethod
+    def _get_backend_meas_freq(backend_data: Optional[BackendData], qubit: int) -> float:
```

Comment:
Do you need this method? I don't think it really improves the readability. Also, [QubitSpectroscopy](https://github.com/Qiskit/qiskit-experiments/blob/1bac9cdd202d8c33e8753ed0bbafdebfe33b67cf/qiskit_experiments/library/characterization/qubit_spectroscopy.py#L27) experiment doesn't implement this mechanism (we need to specify absolute frequency for this experiment). This is bit out of scope, but we should unify the behavior of these experiment, otherwise it's really confusing.

Reply:
Yes, this `absolute` feature adds some complexity. Perhaps it is worth it to have the final results be in terms of absolute frequency though.

Regarding needing this method, what alternative would you prefer? The main purpose of this method is to put in one place the error checking related to retrieving the drive_freq which is currently needed in two places -- `__init__` and `_backend_center_frequency` (called in `circuits`). In `__init__` it shifts the default frequency range based on the meas_freq value so that the frequencies are stored in absolute terms and then in `circuits` it shifts the frequencies back to relative for passing to `shift_frequency`. The class could instead store the frequencies as originally configured along with `absolute` and only do the `drive_freqs` look up at `circuits` time. One unfortunate complexity is that there are three cases -- relative frequencies (passed or using default range), absolute with passed frequencies, and absolute without passed frequencies, but that is not too bad. I think we could change to this behavior without too many backwards compatibility issues for other `Spectroscopy` subclasses. Maybe it is getting to be too big a change to do in this PR though?

Reply:
> retrieving the drive_freq which is currently needed in two places

Okey, I missed the second one. This justifies the need for dedicated method. 

> I think we could change to this behavior without too many backwards compatibility issues for other Spectroscopy subclasses. Maybe it is getting to be too big a change to do in this PR though?

Yes. Let's keep the logical change minimum for this PR. Let's update `Spectroscopy` in a follow-up PR.